documentation/create_overview_doc/createdoc.sh

```diff
@@ -111,419 +82,215 @@ function read_skip_list() {
-    if [[ -n "$mermaid_diagram" && "$mermaid_diagram" =~ [^[:space:]] ]]; then 
-        echo -e "## Flow Diagram [via mermaid]\n\`\`\`mermaid\n$mermaid_diagram\n\`\`\`" >> "$markdown_documentation_file"
-    fi 
 
-    echo -e "Documentation saved to $markdown_documentation_file\n\n"
+    # Generate the flow chart for the module
+    generate_flow_chart "$path_to_module" "$documentation_directory/flowmaps"
+
+    # Add link to the flow chart in the documentation
+    local flow_chart_name=$(basename "$path_to_module" | cut -f 1 -d '.')
+    echo -e "\n![Flow Chart of $flow_chart_name](./flowmaps/${flow_chart_name}_FlowChart.png)\n" >> "$markdown_documentation_file"
+
+    echo -e "Documentation saved to $markdown_documentation_file\n\n" >&2
 }
 
-function extract_module_names_and_associated_objectives_then_call_bito() {
-    local filename=$1
-    local prompt_file_path=$2
-    local lines=()
-    local current_module=""
-    local current_objectives=""
-    local capture_objectives=false
-    local combined_output=""
-    local attempt=1
-    local MAX_RETRIES=5
-    local RETRY_DELAY=10 # seconds
-    local bito_output
-
-    echo -e "Extracting module names and objectives from file: $filename\n" >&2
-
-    # Read the file line by line into the array
-    while IFS= read -r line; do
-        lines+=("$line")
-    done < "$filename"
-
-    # Loop through the lines to process them
+function extract_module_details_and_call_bito() {
+    local filename=$1 prompt_file_path=$2 lines=() current_module="" current_objectives="" current_operational_sequence="" capture_objectives=false capture_operational_sequence=false combined_output=""
+
+    echo "Compiling objectives and details for module in $filename..." >&2
+
+    while IFS= read -r line; do lines+=("$line"); done < "$filename"
+
     for line in "${lines[@]}"; do
         if [[ $line =~ ^##\ Module:\ (.*) ]]; then
             if [[ -n $current_module ]]; then
-                echo "Processing module: $current_module with objectives" >&2
-                combined_output+="Module: $current_module\n---\nPrimary Objectives:\n$current_objectives\n\n"
+                echo "Processing module: $current_module with objectives and operational sequence" >&2
+                combined_output+="Module: $current_module\n---\nPrimary Objectives:\n$current_objectives\nOperational Sequence:\n$current_operational_sequence\n\n"
             fi
             current_module=${BASH_REMATCH[1]}
-            current_objectives=""
-            capture_objectives=false
+            current_objectives="" current_operational_sequence=""
+            capture_objectives=false capture_operational_sequence=false
         elif [[ $line =~ ^-\ \*\*Primary\ Objectives\*\*: ]]; then
             capture_objectives=true
+            capture_operational_sequence=false
             current_objectives+=$(echo $line | sed 's/.*\*\*Primary Objectives\*\*: //')$'\n'
```

Comment:
**Issue**: The retry logic function 'call_bito_with_retry' mentions handling temporary failures for 'bito' but does not account for Code2Flow commands. <br> **Fix**: Adapt the retry logic to include Code2Flow commands or ensure that similar resilience mechanisms are in place for all external tool dependencies. <br> **Code Suggestion**: 
 ```
 function call_bito_with_retry() {
     echo "Initiating Bito command with retry for prompt file $2..." >&2
     local input_text=$1 prompt_file_path=$2 error_message=$3
     local attempt=1 MAX_RETRIES=5 RETRY_DELAY=10 output
     while [ $attempt -le $MAX_RETRIES ]; do
         echo "Attempt $attempt: Calling Bito for prompt '$prompt_file_path'..." >&2
         output=$(echo -e "$input_text" | bito -p "$prompt_file_path")
         local ret_code=$?
         if [ $ret_code -ne 0 ]; then
             echo "Attempt $attempt: $error_message Retrying in $RETRY_DELAY seconds..." >&2
             sleep $RETRY_DELAY
             ((attempt++))
         else
             echo "Bito call successful on attempt $attempt." >&2
             echo "$output"
             update_token_usage "$input_text" "$output"
             return 0
         fi
     done
     echo "Failed to call Bito after $MAX_RETRIES attempts. $error_message" >&2
     return 1
 }
 ```